File: Chapter 05/JS/section_5_4_server_monitor/after/core/ports/alertPort.js
/**
 * PRIMARY PORT (Driven).
 * Since JavaScript lacks native interfaces, we use this base class 
 * to define the 'Contract'. If an Adapter fails to implement sendAlert, 
 * the application will throw a clear architectural error.
 */
class AlertPort {
    /**
     * Sends an alert message to an external destination.
     * @param {string} message 
     */
    sendAlert(message) {
        throw new Error("Method 'sendAlert(message)' must be implemented by the Adapter.");
    }
}

module.exports = AlertPort;

File: Chapter 05/Python/section_5_4_server_monitor/after/core/ports/alert_port.py
from abc import ABC, abstractmethod

class AlertPort(ABC):
    """
    PRIMARY PORT (Driven).
    Using Python's Abstract Base Classes to define the 'Contract'.
    Any adapter that inherits from this MUST implement the send_alert method,
    or Python will throw an instantiation error.
    """
    
    @abstractmethod
    def send_alert(self, message: str) -> None:
        """Sends an alert message to an external destination."""
        pass